context.go

```diff
@@ -76,6 +77,27 @@ func NewContext(opt ...ContextOption) *Context {
 	return ctx
 }
 
+func NewContextFromSnapshot(iso *Isolate, snapshot_index int) (*Context, error) {
```

Comment:
For all of the new APIs added in this PR, we should have some sort of documentation/comments above each explaining them.